src/main/java/teammates/logic/core/FeedbackSessionsLogic.java

```diff
@@ -845,15 +845,75 @@ public FeedbackSessionResultsBundle getFeedbackSessionResultsForStudent(
 
         return getFeedbackSessionResultsSummaryInSectionAsCsv(
-                feedbackSessionName, courseId, userEmail, null, filterText, isMissingResponsesShown, isStatsShown);
+                feedbackSessionName, courseId, userEmail, null, filterText,
+                isMissingResponsesShown, isStatsShown, questionNumber);
     }
 
     public String getFeedbackSessionResultsSummaryInSectionAsCsv(
             String feedbackSessionName, String courseId, String userEmail,
+            String section, String filterText, boolean isMissingResponsesShown, boolean isStatsShown, String questionNumber)
+            throws EntityDoesNotExistException, ExceedingRangeException {
+
+        long indicatedRange = section == null ? 2000 : -1;
+        FeedbackSessionResultsBundle results = getFeedbackSessionResultsForInstructorInSectionWithinRangeFromView(
+                feedbackSessionName, courseId, userEmail, section,
+                indicatedRange, Const.FeedbackSessionResults.QUESTION_SORT_TYPE);
+
+        if (!results.isComplete) {
+            throw new ExceedingRangeException(ERROR_NUMBER_OF_RESPONSES_EXCEEDS_RANGE);
+        }
+        // sort responses by giver > recipient > qnNumber
+        Collections.sort(results.responses,
+                results.compareByGiverRecipientQuestion);
+
+        StringBuilder exportBuilder = new StringBuilder(100);
+
+        exportBuilder.append(String.format("Course,%s",
+                             SanitizationHelper.sanitizeForCsv(results.feedbackSession.getCourseId())))
+                     .append(Const.EOL)
+                     .append(String.format("Session Name,%s",
+                             SanitizationHelper.sanitizeForCsv(results.feedbackSession.getFeedbackSessionName())))
+                     .append(Const.EOL);
+
+        if (section != null) {
+            exportBuilder.append(String.format("Section Name,%s", SanitizationHelper.sanitizeForCsv(section)))
+                         .append(Const.EOL);
+        }
+
+        exportBuilder.append(Const.EOL).append(Const.EOL);
+
+        Set<Entry<FeedbackQuestionAttributes, List<FeedbackResponseAttributes>>> entrySet =
+                results.getQuestionResponseMap().entrySet();
+
+        if (filterText != null && !filterText.isEmpty()) {
+            entrySet = filterQuestions(entrySet, filterText.toLowerCase());
+        }
+        if (questionNumber == null) {
+            for (Map.Entry<FeedbackQuestionAttributes, List<FeedbackResponseAttributes>> entry : entrySet) {
+                exportBuilder.append(getFeedbackSessionResultsForQuestionInCsvFormat(
+                        results, entry, isMissingResponsesShown, isStatsShown));
+            }
+            return exportBuilder.toString();
+        } else {
+            for (Map.Entry<FeedbackQuestionAttributes, List<FeedbackResponseAttributes>> entry : entrySet) {
+                if (Integer.toString(entry.getKey().questionNumber).equals(questionNumber)) {
```

Comment:
As aforementioned, pruning the data here do not reduce the database query time, which I think is the bottleneck of the process. You might want to also improve on db performance by implementing a function to get session for only the specified question.